tools/targets.py

```diff
@@ -172,20 +178,35 @@ def __getattr__(self, attrname):
         return v
 
     # Add one or more new target(s) represented as a Python dictionary in 'new_targets'
-    # It is an error to add a target with a name that already exists.
+    # It it an error to add a target with a name that exists in "targets.json"
+    # However, it is OK to add a target that was previously added via "add_py_targets"
+    # (this makes testing easier without changing the regular semantics)
     @staticmethod
     def add_py_targets(new_targets):
         crt_data = Target.get_json_target_data()
+        # First add all elemnts to the internal dictionary
         for tk, tv in new_targets.items():
-            if crt_data.has_key(tk):
+            if crt_data.has_key(tk) and (not tk in Target.__py_targets):
                 raise Exception("Attempt to add target '%s' that already exists" % tk)
-            # Add target data to the internal target dictionary
             crt_data[tk] = tv
-            # Create the new target and add it to the relevant data structures
+            Target.__py_targets.add(tk)
+        # Then create the new instances and update global variables if needed
```

Comment:
@bogdanm can you confirm whether this is valid? Originally this code came from your fork.